iml-agent/src/action_plugins/stratagem/action_filesync.rs

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+extern crate futures;
+extern crate tokio;
+
+use crate::agent_error::{ImlAgentError, RequiredError};
+use crate::lustre::search_rootpath;
+use futures::{
+    future::{self, TryFutureExt},
+    stream::{self, StreamExt, TryStreamExt},
+};
+use iml_wire_types::{FidError, FidItem};
+use liblustreapi::LlapiFid;
+use std::{collections::HashMap, path::PathBuf};
+use tokio::fs;
+use tokio::process::Command;
+
+struct Work {
+    src_file: String,
+    dest_file: String,
+}
+
+async fn actually_rsync(work: &Work) -> Result<(), std::io::Error> {
+    tracing::error!("running {}", work.src_file);
+    let output = Command::new("rsync")
+        .arg("-a")
+        .arg("-t")
+        .arg("-X")
+        .arg(&work.src_file)
+        .arg(&work.dest_file)
+        .output();
+
+    let output = output.await?;
+
+    tracing::debug!("done {} {}", work.src_file, output.status);
+
+    Ok(())
+}
+
+async fn do_rsync(work_list: &Vec<Work>) -> Result<Vec<FidError>, ImlAgentError> {
+    let result = Vec::new();
+
+    stream::iter(work_list)
+        .map(|xs| Ok::<_, std::io::Error>(xs))
+        .try_for_each_concurrent(100, |fids| {
+            actually_rsync(fids)
+                .or_else(|e| {
+                    println!("error: {}", e);
+                    future::ok(())
+                })
+                .map_ok(|_| println!("done - actually"))
+        })
+        .await?;
+
+    Ok(result)
+}
+
+async fn archive_fids(
+    llapi: LlapiFid,
+    dest_src: &String,
+    fid_list: Vec<FidItem>,
+) -> Result<Vec<FidError>, ImlAgentError> {
+    let mut result = Vec::new();
+
+    let mut rsync_list: Vec<Work> = Vec::new();
+
+    for fid in fid_list {
+        let res = llapi.fid2path(&fid.fid);
+
+        if res.is_err() {
+            tracing::error!("llapi: failed on fid {}", &fid.fid);
+            continue;
+        }
+
+        let fid_path = res.ok().unwrap();
```

Comment:
```suggestion
        let fid_path = match llapi.fid2path(&fid.fid) {
            Ok(x) => x,
            Err(e) => {
                tracing::error!("llapi: failed on fid {}", &fid.fid);
                continue;
            }
        };
```

Equivalent and avoids the `unwrap`.

Reply:
can't continue from a match :/